src/server-extension/resolvers/VideosResolver/index.ts

```diff
@@ -134,30 +135,25 @@ export class VideosResolver {
 
     let connectionQuerySql: string
 
+    connectionQuerySql = extendClause(
+      connectionQuery.sql,
+      'WHERE',
+      `"video"."id" IN (${ids.map((id) => `'${id}'`).join(', ')})`,
+      'AND'
+    )
+
     const hasPeriodDaysArgAndIsOrderedByViews =
       args.periodDays &&
       (args.orderBy.find((orderByArg) => orderByArg === 'viewsNum_DESC') ||
         args.orderBy.find((orderByArg) => orderByArg === 'viewsNum_ASC'))
 
     if (hasPeriodDaysArgAndIsOrderedByViews) {
-      connectionQuerySql = extendClause(
-        connectionQuery.sql,
-        'FROM',
-        `JOIN unnest('{${ids
-          .map((id) => `"${id}"`)
-          .join(', ')}}'::text[]) WITH ORDINALITY t(id, ord) USING ("id")`,
-        ''
-      )
-
-      connectionQuerySql = connectionQuerySql.replace('"video"."views_num" DESC', 't.ord')
-      connectionQuerySql = connectionQuerySql.replace('"video"."views_num" ASC', 't.ord DESC')
-    } else {
-      connectionQuerySql = extendClause(
-        connectionQuery.sql,
-        'WHERE',
-        `"video"."id" IN (${ids.map((id) => `'${id}'`).join(', ')})`,
-        'AND'
-      )
+      const arrayPosition = `array_position(
+        array[${ids.map((id) => `'${id}'`).join(', ')}],
+        video.id  
+      ) ASC`
```

Comment:
I think when replacing `"video"."views_num" DESC` you'd want to use `array_position(...) ASC` and for `"video"."views_num" ASC` - `array_position(...) DESC`

So that for example:
```graphql
{
  mostViewedVideosConnection(
    orderBy: viewsNum_ASC,
    limit: 2,
    periodDays:30
  ) {
    # ...
}
```

Means:
- Pick 2 of the most viewed videos in the last 30 days
- Order them by number of views they had, ascending

This is also how it works currently if you don't specify `periodDays` at all